astropy/io/misc/tests/test_hdf5.py

```diff
@@ -823,3 +838,17 @@ def test_overwrite_serialized_meta():
         t2 = Table.read(out, path='data')
         assert all(t3 == t2)
         assert t3.info(out=None) == t2.info(out=None)
+
+
+@pytest.mark.skipif('not HAS_H5PY')
+def test_read_write_tilde_path(tmpdir, home_is_tmpdir):
+    test_file = os.path.join('~', 'test.hdf5')
+    t1 = Table()
+    t1.add_column(Column(name='a', data=[1, 2, 3]))
+
+    t1.write(test_file, path='tilde')
```

Comment:
What is the `path='tilde'` for?

Reply:
Confusingly in this context, `path` is the argument for the HDF5 writer to specify the name of the table within the HDF5 structure.

Reply:
Indeed, multiple overloaded words in this line. I'll change the value given for `path` so that `path='tilde'` doesn't look like it might be an option flag enabling tilde-expansion